libpromises/matching.c

```diff
@@ -292,7 +292,7 @@ void EscapeSpecialChars(const char *str, char *strEsc, int strEscSz, char *noEsc
                 break;
             }
 
-            strcat(strEsc, noEscSeq);
+            strlcat(strEsc, noEscSeq, strEscSz);
```

Comment:
`strlcat()` with no checking is a sign of bad code - patching stuff with no real understanding. Here this is clearly not the case, checks are quite rigorous above. I woudn't write it like that, I don't care deeply though so I'm cherry-picking this.